R/mod_packageDependencies.R

```diff
@@ -0,0 +1,385 @@
+      }
+    }, ignoreInit = TRUE)
+    
+    m_id <- reactiveVal()
+    
+    pkg_df <- eventReactive({selected_pkg$name(); tabready(); input$toggle_score}, {
+
+      req(!rlang::is_empty(selected_pkg$name()))
+      req(selected_pkg$name() != "-")
+      req(tabready() == 1L)
+
+      req(lastpkg() != selected_pkg$name() | (is.null(assess_dependencies()) || input$toggle_score != assess_dependencies() ))
+      if( lastpkg() != selected_pkg$name()) req(input$toggle_score == FALSE)
+      
+      # start the progress message as soon as possible.
+      m_id(cli::cli_progress_message("Compiling dependency info...", .auto_close = FALSE))
+      
+      pkginfo <- depends() %>%  
+        as_tibble() %>% 
+        mutate(package = stringr::str_replace(package, "\n", "")) %>% 
+        mutate(name = stringr::str_extract(package, "\\w+"))
+
+      # drop any Base R packages from list, unless we ony have 1 row
+      pkginf2 <- pkginfo %>% 
+        filter(!name %in% c(rownames(installed.packages(priority="base")))) 
+      if (nrow(pkginf2) > 0) pkginfo <- pkginf2
+      
+      pkg_name <- ""
+      cl_id <- cli::cli_progress_bar("Assessing Package: ", 
+                                     type = "iterator",
+                                     format = "{pkg_name} {cli::pb_percent} | ETA: {cli::pb_eta}",
+                                     total = nrow(pkginfo))
+      
+      purrr::map_df(pkginfo$name, ~get_versnScore(.x, cl_id, input$toggle_score)) %>% 
+        right_join(pkginfo, by = "name") %>% 
+        select(package, type, name, version, score)
+
+    }, ignoreInit = TRUE) 
+    
+    observeEvent(pkg_df(), {
+      cli::cli_progress_done(id = m_id())
+      cli::cli_progress_done()
+      assess_dependencies(input$toggle_score) # remember last input$toggle_score value
+      lastpkg(isolate(selected_pkg$name()))
+    }) 
+    
+    data_table <- eventReactive(pkg_df(), {
+      cbind(pkg_df(), 
+            data.frame(
+              Actions = shinyInput(actionButton, nrow(pkg_df()),
+                        'button_',
+                        size = "xs",
+                        style='height:24px; padding-top:1px;',
+                        label = icon("arrow-right", class="fa-regular", lib = "font-awesome"),
+                        onclick = paste0('Shiny.setInputValue(\"' , ns("select_button"), '\", this.id)')
+              )
+            )
+      )
+    })
+    
+    # dd <- eventReactive(pkg_df(), {
+    #   x <- deepdep::deepdep(selected_pkg$name(), depth = 2, dependency_type = c("Depends", "Imports", "LinkingTo"))
+    #   if (nrow(x) == 0) {
+    #     x <- data.frame(origin = selected_pkg$name(),
+    #                     name = pull(pkg_df()[1, 3]), version = NA_character_, type = "Imports",
+    #                     origin_level = 0L, dest_level = 1L)
+    #     attributes(x)$class <- c("deepdep", "data.frame")
+    #     attributes(x)$package_name <- selected_pkg$name()
+    #   }
+    #   return(x)
+    # })
```

Comment:
This code should be deleted. We don't want it on `dev`. If you want to save it in another branch, that could be useful to have for later.